asapdiscovery-ml/asapdiscovery/ml/tests/test_inference.py

```diff
@@ -107,3 +106,22 @@ def test_gatinference_predict_dataset_equal(
     assert_allclose(output1, output_infds_1)
     assert_allclose(output2, output_infds_2)
     # everything is the same :)
+
+
+def test_gatinference_predict_from_smiles_dataset_equal(
+    weights_yaml, test_data, test_inference_data
+):
+    inference_cls = asapdiscovery.ml.inference.GATInference(
+        "gatmodel_test", weights_yaml
+    )
+    g1, g2, _ = test_data
+    g1_infds, g2_infds, gids = test_inference_data
+    # same data different smiles order
+    assert inference_cls is not None
+    smiles = list(gids.smiles_dict.keys())
+    assert len(smiles) == 2
+    s1 = smiles[0]
+    s2 = smiles[1]
+    output1 = inference_cls.predict(gids[s1])
+    output2 = inference_cls.predict(gids[s2])
+    assert_allclose(output1, output2)
```

Comment:
Are we testing that predicted values are different from each other? Sometimes with poorly trained models they end up learning to predict the mean label (because they're not able to learn some form of inference so predicting the mean minimizes the cost fn) of the training set and will consistently predict that value.

Reply:
I can add a test for two distinct smiles and make sure they are different. I think thats a good idea.